endpoints/openrtb2/amp_auction_test.go

```diff
@@ -1359,3 +1359,65 @@ func AmpObjectTestSetup(t *testing.T, inTagId string, inStoredRequest json.RawMe
 	)
 	return &actualAmpObject, endpoint
 }
+
+func TestAmpAuctionResponseHeaders(t *testing.T) {
+	testCases := []struct {
+		description         string
+		requestURLArguments string
+		expectedStatus      int
+		expectedHeaders     func(http.Header)
```

Comment:
For my own understanding, what is happening here with `func(http.Header)`? I haven't seen this syntax with `func(something)` and see it in a couple of places here in this PR.

Reply:
Good question. I am defining a function pointer variable. This is the same format as a normal function header, but I'm omitting the variable name since it doesn't matter in this context. In the test cases, I'm defining anonymous methods and need to provide a variable name for use in the body.

I'm doing this here so that I can use the `.Set()` method for adding headers to the map. Since it is a really just a map, I could get away with a map literal.. but there is something formatting performed by the headers class and I want to be sure I'm using the exact same code paths for the test. I could also use a map and then loop through in making calls to `.Set()`, but I thought this was easier and more direct. It's an implementation choice and I wouldn't comment on anyone using an alternate approach. It's just that this is what looks good / makes sense to me.